src/vector.js

```diff
@@ -26,26 +27,147 @@ class Vector {
       this.x = x;
       this.y = y;
     }
+    // @endif
   }
 
   /**
-   * Return a new Vector whose value is the addition of the current Vector and the passed in Vector. If `dt` is provided, the result is multiplied by the value.
+   * Calculate the addition of the current vector with the passed in vector.
    * @memberof Vector
    * @function add
    *
    * @param {Vector|{x: number, y: number}} vector - Vector to add to the current Vector.
-   * @param {Number} [dt=1] - Time since last update.
    *
    * @returns {Vector} A new Vector instance.
    */
-  add(vec, dt = 1) {
+  add(vec) {
     return new Vector(
-      this.x + (vec.x || 0) * dt,
-      this.y + (vec.y || 0) * dt,
+      this.x + vec.x,
+      this.y + vec.y,
       this
     );
   }
 
+  // @ifdef VECTOR_SUBTRACT
+  /**
+   * Calculate the subtraction of the current vector with the passed in vector.
+   * @memberof Vector
+   * @function subtract
+   *
+   * @param {Vector|{x: number, y: number}} vector - Vector to subtract from the current Vector.
+   *
+   * @returns {Vector} A new Vector instance.
+   */
+   subtract(vec) {
+    return new Vector(
+      this.x - vec.x,
+      this.y - vec.y,
+      this
+    );
+  }
+  // @endif
+
+  // @ifdef VECTOR_SCALE
+  /**
+   * Calculate the multiple of the current vector by a value.
+   * @memberof Vector
+   * @function scale
+   *
+   * @param {Number} value - Value to scale the current Vector.
+   *
+   * @returns {Vector} a new Vector instance.
+   */
+  scale(value) {
+    return new Vector(
+      this.x * value,
+      this.y * value
+    );
+  }
+  // @endif
+
+  // @ifdef VECTOR_NORMALIZE
+  // @ifdef VECTOR_LENGTH
+  /**
+   * Calculate the normalized value of the current vector. Requires the Vector (length)[/api/vector#length] function.
+   * @memberof Vector
+   * @function normalize
+   *
+   * @returns {Vector} a new Vector instance.
+   */
+  // @see https://github.com/jed/140bytes/wiki/Byte-saving-techniques#use-placeholder-arguments-instead-of-var
+  normalize(length = this.length()) {
+    return new Vector(
```

Comment:
I always wonder if we should treat null vectors differently. A null vector would produce an `(NaN, NaN)` vector. I assume that this problem is known and the code is intentional. My math library is essentially doing the same. :-)

Reply:
I've mostly opted out of error checking in favor or smaller size. There's sometimes I do it if when functionality is required to work (such as in Pool).